python/dgllife/model/model_zoo/potentialnet.py

```diff
@@ -0,0 +1,223 @@
+    Added a linear layer at the end as described in PotentialNet stage 1 & 2.
+
+    Parameters
+    ----------
+    in_feats : int
+        Input feature size; i.e, the number of dimensions of :math:`x_i`.
+    out_feats : int
+        Output feature size from GatedGraphConv; i.e., the number of dimensions of :math:`h_i^{(t+1)}`,
+        equivalent to the input feature size to the linear layer,
+    f_gather: int
+        Output feature size from the linear layer,
+    n_steps : int
+        Number of recurrent steps; i.e, the :math:`t` in the above formula.
+    n_etypes : int
+        Number of edge types.
+    dropout: float
+        Amount of dropout applied after GatedGraphConv and before the linear layer.
+    bias : bool
+        If True, adds a learnable bias to the output. Default: ``True``.
+    """
+    def __init__(self,
+                in_feats,
+                out_feats,
+                f_gather,
+                n_steps,
+                n_etypes,
+                dropout,
+                bias=True):
+        super(Customized_GatedGraphConv, self).__init__()
+        self._in_feats = in_feats
+        self._out_feats = out_feats
+        self._n_steps = n_steps
+        self._n_etypes = n_etypes
+        self.linears = nn.ModuleList(
+            [nn.Linear(out_feats, out_feats) for _ in range(n_etypes)]
+        )
+        self.gru = nn.GRUCell(out_feats, out_feats, bias=bias)
+        self.dropout = nn.Dropout(p=dropout)
+        self.i_nn = nn.Linear(in_features=(in_feats + out_feats), out_features=f_gather)
+        self.j_nn = nn.Linear(in_features=out_feats, out_features=f_gather)
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        gain = init.calculate_gain('relu')
+        self.gru.reset_parameters()
+        for linear in self.linears:
+            init.xavier_normal_(linear.weight, gain=gain)
+            init.zeros_(linear.bias)
+
+    def set_allow_zero_in_degree(self, set_value):
+        self._allow_zero_in_degree = set_value
+
+    def forward(self, graph, feat):
+        """
+
+        Description
+        -----------
+        Compute Gated Graph Convolution layer.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            The graph, with graph.ndata['h'] being the input feature of shape :math:`(N, D_{in})` where :math:`N`
+            is the number of nodes of the graph and :math:`D_{in}` is the
+            input feature size;
+            And graph.edata['e'] one-hot encodes the edge types
+
+        Returns
+        -------
+        torch.Tensor
+            The output feature of shape :math:`(N, D_{out})` where :math:`D_{out}`
+            is the output feature size.
+        """
+        with graph.local_scope():
+            assert graph.is_homogeneous, \
+                "not a homogeneous graph; convert it with to_homogeneous " \
+                "and pass in the edge type as argument"
+            assert  graph.edata['e'].shape[1] <= self._n_etypes, \
```

Comment:
Wrong empty space

Reply:
Sorry could you be more specific?

Reply:
The empty space between "assert" and "graph.edata['e']" does not seem to be correct.